crypto/asn1/tasn_fre.c

```diff
@@ -89,7 +89,8 @@ static void asn1_item_embed_free(ASN1_VALUE **pval, const ASN1_ITEM *it,
 
     case ASN1_ITYPE_NDEF_SEQUENCE:
     case ASN1_ITYPE_SEQUENCE:
-        if (asn1_do_lock(pval, -1, it) > 0)
+        count = asn1_do_lock(pval, -1, it);
```

Comment:
I add it to not reuse 'i'.
The test is changed, to continue the free-ing code only when ref-count is 0 only, 
and not on error ( < 0 ) ...